### ISSUE TITLE
Adding fields that are not present when saving

### ISSUE THREAD

**Issue by joetidee** [jagi/meteor-astronomy#356]
I'm finding that when saving a document, if some of the fields that are defined in the model are not present at the point of saving the document, they are still inserted into the docuemnt (with values such as null). I noticed that you can specify exactly the fields that are to be saved in the save() function, hoever, that becomes cumbersome having to add this seperate logic.

Is there a way to run save() and any fields that are not present (and not defined as required) are not inserted into the document?

For example, in my model I have category defined:

```
category: {
            type: ['object'],
            default: function() {
                return [{
                    id: 1,
                    name: '-'
                }];
            },
            optional: true
        }
```

This field is optional, so if the form that the user submits doesn't have category selected, which in turn means that the model sent to my Meteor.method doesn't have a category set, then some logic needs triggering when the document is inserted. Perhaps it could be that if a default is set for the field definition in the model, then this should be inserted into the document. If the field is not set (and is `optional`), then it should **not** be inserted.

**Comment by lukejagodzinski** [comment]
@joetidee you're mixing two syntaxes from Astronomy 1.0 and 2.0. You don't specify type as `['object']` it's not correct way of doing it in any version. Please read documentation first before asking any question. If it goes about not provided values for a fields in 1.x it will fill these values with nulls, however in 2.0 it's leaving them undefined. The way it works in 2.0 is much more intuitive and probably more correct.

**Comment by joetidee** [comment]
Do you know when v2 will be released?

**Comment by lukejagodzinski** [comment]
I hope that I will finish it until the end of this month

**Comment by lukejagodzinski** [comment]
I'm preparing to release 2.0 so I'm closing this issue for now. I want to clear out all issues that got fixed in 2.0
